WMFData/Sources/WMFData/Data Controllers/Year In Review/WMFYearInReviewDataController.swift

```diff
@@ -288,6 +285,97 @@ public class WMFYearInReviewDataController {
         return results
     }
     
+    private func populateReadingSlide(report: CDYearInReviewReport, backgroundContext: NSManagedObjectContext) throws {
+        
+        guard let iosFeatureConfig = developerSettingsDataController.loadFeatureConfig()?.ios.first,
+              let yirConfig = iosFeatureConfig.yir(yearID: targetConfigYearID) else {
+            return
+        }
+        
+        guard let dataPopulationStartDate = yirConfig.dataPopulationStartDate,
+              let dataPopulationEndDate = yirConfig.dataPopulationEndDate else {
+            return
+        }
+        
+        let pageViewsDataController = try WMFPageViewsDataController(coreDataStore: coreDataStore)
+        let pageViewCounts = try pageViewsDataController.fetchPageViewCounts(startDate: dataPopulationStartDate, endDate: dataPopulationEndDate, moc: backgroundContext)
+        
+        guard let slides = report.slides as? Set<CDYearInReviewSlide> else {
+            return
+        }
+        
+        for slide in slides {
+            
+            guard let slideID = slide.id else {
+                continue
+            }
+            
+            switch slideID {
+            case WMFYearInReviewPersonalizedSlideID.readCount.rawValue:
+                let encoder = JSONEncoder()
+                slide.data = try encoder.encode(pageViewCounts.count)
+                
+                if pageViewCounts.count > 5 {
+                    slide.display = true
+                }
+                
+                slide.evaluated = true
+            default:
+                break
+            }
+        }
+        
+        try coreDataStore.saveIfNeeded(moc: backgroundContext)
+    }
+    
+    private func fetchEditCount(username: String, project: WMFProject?) async throws -> Int {
+        
+        guard let iosFeatureConfig = developerSettingsDataController.loadFeatureConfig()?.ios.first,
+              let yirConfig = iosFeatureConfig.yir(yearID: targetConfigYearID) else {
+            throw WMFYearInReviewDataControllerError.missingRemoteConfig
+        }
+        
+        guard let dataPopulationStartDate = yirConfig.dataPopulationStartDate,
```

Comment:
Let's address this TODO: here now too. The config already has a string-ified start and end date, so change this to `let dataPopulationStartDateString = yirConfig.dataPopulationStartDateString,
let dataPopulationEndDateString = yirConfig.dataPopulationEndDateString`